boot/boot.go

```diff
@@ -395,3 +395,41 @@ func SetRecoveryBootSystemAndMode(dev Device, systemLabel, mode string) error {
 	}
 	return bl.SetBootVars(m)
 }
+
+// UpdateManagedBootConfigs updates managed boot config assets if those are
+// present for the ubuntu-boot bootloader. Returns true when an update was
+// carried out.
+func UpdateManagedBootConfigs(dev Device) (updated bool, err error) {
+	if !dev.HasModeenv() {
+		// only UC20 devices use managed boot config
+		return false, nil
+	}
+	if !dev.RunMode() {
+		return false, fmt.Errorf("internal error: boot config can only be updated in run mode")
+	}
+	return updateManagedBootConfigForBootloader(dev, ModeRun)
+}
+
+func updateManagedBootConfigForBootloader(dev Device, mode string) (updated bool, err error) {
+	if mode != ModeRun {
+		return false, fmt.Errorf("internal error: updating boot config of recovery bootloader is not supported yet")
+	}
+
+	opts := &bootloader.Options{
+		Role:        bootloader.RoleRunMode,
+		NoSlashBoot: true,
+	}
+	tbl, err := getBootloaderManagingItsAssets(InitramfsUbuntuBootDir, opts)
```

Comment:
it's slightly odd to use InitramfsUbuntuBootDir nad NoSlashBoot for a runtime update, maybe @anonymouse64 has thoughts on this

Reply:
hmm I guess it's not that odd since we are using NoSlashBoot with the /run/mnt/ubuntu-boot directory, and ubuntu-boot partition does not have the "/boot" layout by definition.

I do wonder why we can't just use "" as the directory and not specify NoSlashBoot ?  afaict this code here is here because that's what composeCommandLine does and this seems to be a parallel of that code, but actually looking back at that code, I don't see why it wouldn't work for that to also use `""` as the bootloader root dir and leave NoSlashBoot false in the run mode case too

Reply:
I'll play with it in a followup.